biojava-structure/src/main/java/org/biojava/nbio/structure/rcsb/PdbIdLists.java

```diff
@@ -0,0 +1,259 @@
+                "  <orgPdbQuery>\n" +
+                "    <version>head</version>\n" +
+                "    <queryType>org.pdb.query.simple.HoldingsQuery</queryType>\n" +
+                "    <description>Holdings : All Structures</description>\n" +
+                "    <experimentalMethod>ignore</experimentalMethod>\n" +
+                "    <moleculeType>ignore</moleculeType>\n" +
+                "  </orgPdbQuery>\n" +
+                " </queryRefinement>\n" +
+                " <queryRefinement>\n" +
+                "  <queryRefinementLevel>1</queryRefinementLevel>\n" +
+                "  <conjunctionType>and</conjunctionType>\n" +
+                "  <orgPdbQuery>\n" +
+                "    <version>head</version>\n" +
+                "    <queryType>org.pdb.query.simple.MacroMoleculeQuery</queryType>\n" +
+                "    <description>Molecule : Gag-Pol polyprotein [A1Z651, O12158, P03355, P03366, P03367, P03369, P04584, P04585, P04586, P04587, P04588, P05896, P05897, P05959, P05961, P0C6F2, P12497, P12499, P18042, P19505 ... ]</description>\n" +
+                "    <macromoleculeName>A1Z651,O12158,P03355,P03366,P03367,P03369,P04584,P04585,P04586,P04587,P04588,P05896,P05897,P05959,P05961,P0C6F2,P12497,P12499,P18042,P19505,P19560,P20875,P24740,P35963,Q699E2,Q70XD7,Q72547,Q7SMT3,Q7SPG9,Q90VT5</macromoleculeName>\n" +
+                "  </orgPdbQuery>\n" +
+                " </queryRefinement>\n" +
+                "</orgPdbCompositeQuery>";
+
+        return postQuery(xml);
+    }
+
+    /** get all Transmembrane proteins
+     *
+     * @return list of PDB IDs
+     * @throws IOException
+     */
+    public static Set<String> getTransmembraneProteins() throws IOException {
+        String xml = "  <orgPdbQuery>\n" +
+                "    <version>head</version>\n" +
+                "    <queryType>org.pdb.query.simple.TreeQuery</queryType>\n" +
+                "    <description>TransmembraneTree Search for root</description>\n" +
+                "    <t>19</t>\n" +
+                "    <n>0</n>\n" +
+                "    <nodeDesc>root</nodeDesc>\n" +
+                "  </orgPdbQuery>";
+
+        return postQuery(xml);
+    }
+
+    public static final String SERVICELOCATION="http://www.rcsb.org/pdb/rest/search";
+
+
+    /** post am XML query (PDB XML query format)  to the RESTful RCSB web service
+     *
+     * @param xml
+     * @return a list of PDB ids.
+     */
+    public static Set<String> postQuery(String xml)
+            throws IOException{
+
+
+
+        URL u = new URL(SERVICELOCATION);
+
+
+        String encodedXML = URLEncoder.encode(xml,"UTF-8");
+
+
+        InputStream in =  doPOST(u,encodedXML);
+
+        Set<String> pdbIds = new TreeSet<String>();
+
+
+        BufferedReader rd = new BufferedReader(new InputStreamReader(in));
+
+        String line;
+        while ((line = rd.readLine()) != null) {
+
+            pdbIds.add(line);
+
+        }
+        rd.close();
```

Comment:
This should be either in a finally or the InputStream should be in a try-with-resources, see https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html. Otherwise resources would leak if IOException is thrown.